src/main/java/budgetbuddy/model/AccountsManager.java

```diff
@@ -104,8 +106,12 @@ public void addAccount(Account toAdd) {
      * @param editedAccount The edited account to replace the target account with.
      */
     public void editAccount(Index toEdit, Account editedAccount) throws AccountNotFoundException {
-        setActiveAccount(toEdit);
         accounts.replace(filteredAccounts.get(toEdit.getZeroBased()), editedAccount);
+        unsetActiveAccount();
+        //editing the account may cause it to no longer be part of the filtered list,
+        //so we have to reset the filtered account list.
+        resetFilteredAccountList();
```

Comment:
since reset actually resets the active account, do you still need to unset here?

Reply:
ah thanks for catching that, changed